src/Illuminate/Pipeline/Pipeline.php

```diff
@@ -182,6 +189,10 @@ protected function carry()
 
                     return $this->handleCarry($carry);
                 } catch (Throwable $e) {
+                    if ($this->onFailure) {
+                        return call_user_func($this->onFailure, $pipe);
```

Comment:
It would be nice to also send the exception to the `onFailure` method, for logging purposed, for example.

I'd also suggest to send it as the first parameter.

Reply:
I don't have any opinion on that, let's see what will Taylor say.

Reply:
@michael-rubel Didn't get it, what's wrong with `call_user_func`?
Could you please be more specific?

Reply:
@rezaamini-ir I meant using of more modern approach `($this->onFailure)($pipe)` instead of `call_user_func`.
Look at this line: https://github.com/michael-rubel/laravel-enhanced-pipeline/blob/main/src/Pipeline.php#L236